Add explicit return to JuliaFormatter and docs

clizbe opened the pull request:
<!--
Thanks for making a pull request to TulipaEnergyModel.jl.
We have added this PR template to help you help us.

Make sure to read the contributing guidelines.

See the comments below, fill the required fields, and check the items.
-->

## Related issues

<!-- We normally work with (i) create issue; (ii) discussion if necessary; (iii) create PR. So, at least one of the following should be true:-->

<!-- Option 1, this closes an existing issue. Fill the number below-->
Closes #796
Closes #940

<!-- Option 2, this is a small fix that arguably won't need an issue. Uncomment below -->
<!--
There is no related issue.
-->

## Checklist

<!-- mark true if NA -->
<!-- leave PR as draft until all is checked -->

- [X] I am following the [contributing guidelines](https://github.com/TulipaEnergy/TulipaEnergyModel.jl/blob/main/docs/src/90-contributing.md)

- [X] Tests are passing
- [X] Lint workflow is passing
- [X] Docs were updated and workflow is passing

clizbe commented:
I'm not sure all the files should be changed with adding 'return.' 
I ran JuliaFormatter using format(".") which might touch files we don't normally format.

clizbe commented on docs/src/91-developer.md:
```diff
@@ -441,8 +442,7 @@ See the file <benchmark/profiling.jl> for an example of profiling code.
 > **Note:**
-> To be able to register, you need to be a member of the organisation TulipaEnergy and have your visibility set to public:<br>
-> ![Screenshot of public members of TulipaEnergy on GitHub](./images/PublicMember.png)
+> To be able to register, you need to be a member of the organisation TulipaEnergy and have your visibility set to public:<br> > ![Screenshot of public members of TulipaEnergy on GitHub](./images/PublicMember.png)
```
Not sure why this changed.

clizbe commented on src/constraints/consumer.jl:
```diff
@@ -19,7 +19,7 @@ function add_consumer_constraints!(model, constraints, graph, sets)
     # - Balance constraint (using the lowest temporal resolution)
     df = filter(:asset => ∈(Ac), constraints[:highest_in_out].indices; view = true)
-    model[:consumer_balance] = [
+    return model[:consumer_balance] = [
```
I wonder if this is correct or these should all be `return nothing`.

abelsiqueira replied:
The JuliaFormatter is technically correct, because these were indeed being returned implicitly. But we don't expect any return, so adding an empty `return` after this is better, in my opinion

abelsiqueira commented on docs/src/91-developer.md:
```diff
@@ -198,6 +198,7 @@ We will try to follow these during development and reviews.
   - List obvious objects, e.g., `using JuMP: @variable`, since `@variable` is obviously from JuMP in this context, or `using Graph: SimpleDiGraph`, because it's a constructor with an obvious name.
   - For other objects inside `Package`, use `using Package: Package` and explicitly call `Package.A` to use it, e.g., `DataFrames.groupby`.
   - List all `using` in <src/TulipaEnergyModel.jl>.
+- Explicitly state what a function will `return`, including `return nothing`.
```
```suggestion
  - Explicitly state what a function will `return`, including `return nothing`.
```

abelsiqueira commented:
Indeed, many places will be changed with this new rule because all functions return something (implicit rule is to return the last thing in a function). This is annoying, but necessary with this change. For most places, we are going to need manual intervention to add and empty `return` before the end to reflect the desired behaviour (assuming the desired behaviour is to return nothing for functions that are not expected to return anything)

clizbe commented:
@abelsiqueira I fixed a lot of them, but the rest I'm not sure if it was an implicit return or returning nothing.
Also I changed the docs to just `return` instead of `return nothing` since it looks like that's what we were using (and in my opinion it looks better).

clizbe commented:
Dangit I dunno why the conflict merger never works for me. I must be skipping a step. -_-

abelsiqueira reviewed (changes requested):
Thanks for the update. I marked a few places, which are the most relevant. The files in `utils/scripts` don't matter, because they won't be run again, so I haven't reviewed whether it should be an empty return or not.

When it's ready to be merged, please mark as ready to review, I'll approve, and we can try to merge at a moment that avoids conflicts (since it's changing a bunch of files)

clizbe commented:
Thanks! Yeah sorry for jumping in and creating conflicts. 
Just got bored and missed coding!
